tests/Aql/SubqueryEndExecutorTest.cpp

```diff
@@ -0,0 +1,295 @@
+  OutputAqlItemRow output{std::move(outputBlock), infos.getOutputRegisters(),
+                          infos.registersToKeep(), infos.registersToClear()};
+  std::tie(state, stats) = testee.produceRows(output);
+  EXPECT_EQ(state, ExecutionState::DONE);
+  EXPECT_EQ(output.numRowsWritten(), 2);
+  ExpectedValues(output, {"[42]", "[34]"}, {});
+}
+
+TEST_F(SubqueryEndExecutorTest, two_input_one_shadowrow_two_irrelevant) {
+  SharedAqlItemBlockPtr outputBlock;
+  SharedAqlItemBlockPtr inputBlock =
+      buildBlock<1>(itemBlockManager, {{42}, {42}, {42}, {42}, {42}},
+                    {{2, 0}, {3, 1}, {4, 2}});
+
+  SingleRowFetcherHelper<false> fetcher(itemBlockManager, inputBlock->size(), false, inputBlock);
+
+  SubqueryEndExecutor testee(fetcher, infos);
+
+  NoStats stats{};
+  ExecutionState state{ExecutionState::HASMORE};
+
+  outputBlock.reset(new AqlItemBlock(itemBlockManager, inputBlock->size(), 1));
+  OutputAqlItemRow output{std::move(outputBlock), infos.getOutputRegisters(),
+                          infos.registersToKeep(), infos.registersToClear()};
+
+  std::tie(state, stats) = testee.produceRows(output);
+  EXPECT_EQ(state, ExecutionState::DONE);
+  EXPECT_EQ(output.numRowsWritten(), 3);
+  ExpectedValues(output, {"[42, 42]", "", ""}, {{1,0}, {2,1}});
+}
+
+TEST_F(SubqueryEndExecutorTest, consume_output_of_subquery_end_executor) {
+  NoStats stats{};
+  ExecutionState state{ExecutionState::HASMORE};
+
+  SharedAqlItemBlockPtr outputBlock;
+  SharedAqlItemBlockPtr inputBlock =
+      buildBlock<1>(itemBlockManager, {{42}, {42}, {42}, {42}, {42}},
+                    {{2, 0}, {3, 1}, {4, 2}});
+
+  SingleRowFetcherHelper<false> fetcher(itemBlockManager, inputBlock->size(), false, inputBlock);
+
+  SubqueryEndExecutor testee(fetcher, infos);
+
+  outputBlock.reset(new AqlItemBlock(itemBlockManager, inputBlock->size(), 1));
+  OutputAqlItemRow output{std::move(outputBlock), infos.getOutputRegisters(),
+                          infos.registersToKeep(), infos.registersToClear()};
+  std::tie(state, stats) = testee.produceRows(output);
+  EXPECT_EQ(state, ExecutionState::DONE);
+  EXPECT_EQ(output.numRowsWritten(), 3);
+
+  //  ExpectedValues(output, { "[42, 42]", "", "" });
+
+  outputBlock = output.stealBlock();
+  inputBlock.swap(outputBlock);
+  SingleRowFetcherHelper<false> fetcher2(itemBlockManager, inputBlock->size(),
+                                         false, inputBlock);
+  SubqueryEndExecutor testee2(fetcher2, infos);
+  outputBlock.reset(new AqlItemBlock(itemBlockManager, inputBlock->size(), 1));
+  OutputAqlItemRow output2{std::move(outputBlock), infos.getOutputRegisters(),
+                           infos.registersToKeep(), infos.registersToClear()};
+  std::tie(state, stats) = testee2.produceRows(output2);
+  EXPECT_EQ(state, ExecutionState::DONE);
+  EXPECT_EQ(output2.numRowsWritten(), 2);
+
+  ExpectedValues(output2, {"[ [42, 42] ]", ""}, {{1,0}});
+}
+
```

Comment:
Tests look good so far.
I would like to so another test, where the outputRegister is NOT part of the shadowRow data, but after it.

E.g. input is a Block with 1 Register.
Output is a block with 2 Registers, Executor should write to register (1) (the last one).
=> We need to assert that the value in Subquery is maintained, and the Array is written in the desired position.

e.g.

```
SharedAqlItemBlockPtr inputBlock =
       buildBlock<1>(itemBlockManager, {{42}, {23}},
                     {{1, 0}}});
```

=>
Outputrow:
`<23, [42]>` non shadow.
I guess this will not work easily with your helper method. (as you need a second value to assert)